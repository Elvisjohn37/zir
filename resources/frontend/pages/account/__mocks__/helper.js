let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
