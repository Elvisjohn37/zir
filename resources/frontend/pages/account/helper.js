export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
