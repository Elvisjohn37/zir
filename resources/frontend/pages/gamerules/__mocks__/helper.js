let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
