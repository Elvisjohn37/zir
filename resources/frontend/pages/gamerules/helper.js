import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
