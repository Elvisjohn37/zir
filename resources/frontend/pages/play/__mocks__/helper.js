let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
