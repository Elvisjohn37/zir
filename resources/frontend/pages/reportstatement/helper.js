import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
