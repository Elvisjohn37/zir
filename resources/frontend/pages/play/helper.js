import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
