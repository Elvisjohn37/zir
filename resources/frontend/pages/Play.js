import React, { useState, useEffect } from 'react';
import { ErrorPage } from 'frontend/components/OnPageError';
import styles from './Play.module';
import AccountBalance from 'frontend/components/AccountBalance';
import { MoneyFormatToggle } from 'frontend/components/MoneyFormat';
import { FormattedMessage } from 'react-intl';
import { LangDataProvider } from 'frontend/components/Language';
import RunningDateTime from './play/RunningDateTime';
import { UserConsumer } from 'frontend/contexts/User';
import Spinner from 'frontend/components/Spinner';
import classnames from 'classnames';
import Announcement from 'frontend/components/Announcement';
import { postMessageHandler, windowResponse } from './play/helper';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContentText from '@mui/material/DialogContentText';
import DialogContent from '@mui/material/DialogContent';
import Button from '@mui/material/Button';
import { isEmpty } from 'frontend/utils/helper';

export default function Play() {
	let [isLoading, setIsLoading] = useState(true);
	let isMobile = windowResponse().data.isMobile;
	let balanceInfo = windowResponse().data.balanceInfo;
	let playableBalance = isEmpty(balanceInfo) ? 0 : balanceInfo.playableBalance;
	let [isZeroBalancePopup, setIsZeroBalancePopup] = useState(playableBalance <= 0);

	useEffect(() => {
		window.addEventListener('message', postMessageHandler, false);

		return () => {
			window.removeEventListener('message', postMessageHandler);
		};
	});

	return windowResponse().result === true ? (
		<LangDataProvider category="gamewindow">
			<UserConsumer>
				{({ userState }) => (
					<>
						{isMobile ? (
							<Announcement viewType={windowResponse().data.productID} isMobile={isMobile} />
						) : (
							<div className={styles.header}>
								{/* <div className={styles.logoContainer}>
										<Image src={logo} className={styles.logoImage} />
									</div> */}
								<div className={styles.headerContent}>
									<div className={styles.topContainer}>
										<div className={styles.runningDateTime}>
											<RunningDateTime className="testRunningDateTime" />
										</div>
									</div>
									<div className={styles.bottomContainer}>
										<div className={styles.announcementWrapper}>
											<Announcement
												className="testAnnouncement"
												viewType={windowResponse().data.productID}
												isMobile={false}
											/>
										</div>
										<div className={styles.balancesWrapper}>
											<div className={styles.maxPayoutWrapper}>
												<div className={styles.maxPayout}>
													<div className={styles.amountText}>
														<span className={styles.amountFigure}>
															<FormattedMessage id="maxPayoutPerBet" />
															{`${userState.currencyCode} `}
															<MoneyFormatToggle
																value={windowResponse().data.maxPayout}
																compactStart={100000000}
															/>
														</span>
													</div>
												</div>
											</div>
											<AccountBalance
												format="{currencyCode} {balance}"
												className={styles.accountBalance}
												withRefreshButton={true}
												isAutoUpdate={true}
												compactStart={100000000}
												initialBalanceInfo={balanceInfo}
											/>
										</div>
									</div>
								</div>
							</div>
						)}
						<div className={classnames([styles.body, isMobile && styles.mobileBody])}>
							<div className={styles.content}>
								{isLoading && <Spinner color="primary" className={styles.iframeSpinner} />}
								{!isZeroBalancePopup && (
									<iframe
										className={styles.gameIframe}
										src={windowResponse().data.url}
										onLoad={() => setIsLoading(false)}
									/>
								)}
							</div>
						</div>
						<Dialog open={isZeroBalancePopup} maxWidth="xs">
							<DialogContent>
								<DialogContentText classes={{ root: styles.zeroBalanceMessage }}>
									<FormattedMessage id="zeroBalance" />
								</DialogContentText>
								<DialogActions>
									<Button
										onClick={() => {
											setIsZeroBalancePopup(false);
										}}
										color="secondary"
										variant="contained"
									>
										<FormattedMessage id="ok" />
									</Button>
								</DialogActions>
							</DialogContent>
						</Dialog>
					</>
				)}
			</UserConsumer>
		</LangDataProvider>
	) : (
		<ErrorPage />
	);
}
