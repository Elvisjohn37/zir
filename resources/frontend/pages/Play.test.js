import React from 'react';
import Play from './Play';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { resetWindowResponse, setWindowResponse, windowResponse, setErrorResponse } from './play/helper';

jest.mock('frontend/components/OnPageError');
jest.mock('frontend/components/AccountBalance');
jest.mock('frontend/components/MoneyFormat');
jest.mock('react-intl');
jest.mock('frontend/components/Language');
jest.mock('./play/RunningDateTime');
jest.mock('frontend/contexts/User');
jest.mock('frontend/components/Spinner');
jest.mock('classnames');
jest.mock('frontend/components/Announcement');
jest.mock('./play/helper');
jest.mock('@mui/material/Dialog');
jest.mock('@mui/material/DialogActions');
jest.mock('@mui/material/DialogContentText');
jest.mock('@mui/material/DialogContent');
jest.mock('@mui/material/Button');
jest.mock('frontend/utils/helper');

describe('<Play />', () => {
	beforeEach(() => {
		resetWindowResponse();
	});

	it('should match snapshot', () => {
		let { asFragment, rerender } = render(<Play />);
		let desktopFragment = asFragment();

		setWindowResponse({ isMobile: true });
		rerender(<Play />);
		let mobileFragment = asFragment();

		expect(desktopFragment).toMatchSnapshot();
		expect(mobileFragment).toMatchSnapshot();
	});

	it('should display message when playable balance is zero', () => {
		let { getByTestId } = render(<Play />);

		expect(getByTestId('Dialog')).not.toHaveAttribute('open');

		cleanup();
		setWindowResponse({ balanceInfo: { playableBalance: 0 } });
		let { getByTestId: getByTestId2 } = render(<Play />);

		expect(getByTestId2('Dialog')).toHaveAttribute('open');
	});

	it('should display game announcement for desktop when device is desktop', () => {
		let { getByTestId } = render(<Play />);
		let announcement = getByTestId('Announcement');

		expect(announcement).toHaveAttribute('ismobile', 'false');
		expect(announcement).toHaveAttribute('viewType', windowResponse().data.productID);
	});

	it('should display game announcement for mobile when device is mobile', () => {
		setWindowResponse({ isMobile: true });
		let { getByTestId } = render(<Play />);
		let announcement = getByTestId('Announcement');

		expect(announcement).toHaveAttribute('ismobile', 'true');
		expect(announcement).toHaveAttribute('viewType', windowResponse().data.productID);
	});

	it('should display live date and time when device is desktop', () => {
		let { getByTestId, queryByTestId, rerender } = render(<Play />);

		expect(getByTestId('RunningDateTime')).toBeInTheDocument();

		setWindowResponse({ isMobile: true });

		rerender(<Play />);

		expect(queryByTestId('RunningDateTime')).not.toBeInTheDocument();
	});

	it('should display max payout when  device is desktop ', () => {
		let { getByTestId, queryByTestId, rerender } = render(<Play />);
		let maxPayout = getByTestId('MoneyFormatToggle');

		expect(maxPayout).toBeInTheDocument();
		expect(maxPayout).toHaveAttribute('value', windowResponse().data.maxPayout);

		setWindowResponse({ isMobile: true });

		rerender(<Play />);

		expect(queryByTestId('MoneyFormatToggle')).not.toBeInTheDocument();
	});

	it('should display live account balance when device is desktop', () => {
		let { getByTestId, queryByTestId, rerender } = render(<Play />);
		let accountBalance = getByTestId('AccountBalance');

		expect(accountBalance).toBeInTheDocument();
		expect(accountBalance).toHaveAttribute('initialbalanceinfo', JSON.stringify(windowResponse().data.balanceInfo));
		expect(accountBalance).toHaveAttribute('isautoupdate', 'true');

		setWindowResponse({ isMobile: true });

		rerender(<Play />);

		expect(queryByTestId('AccountBalance')).not.toBeInTheDocument();
	});

	it('should display loading indicator when loading game', () => {
		let { getByTestId, queryByTestId, baseElement } = render(<Play />);

		expect(getByTestId('Spinner')).toBeInTheDocument();

		fireEvent.load(baseElement.querySelector('iframe'));

		expect(queryByTestId('Spinner')).not.toBeInTheDocument();
	});

	it('should load game', () => {
		let { baseElement } = render(<Play />);
		let gameIframe = baseElement.querySelector('iframe');

		expect(gameIframe).toBeInTheDocument();
		expect(gameIframe).toHaveAttribute('src', windowResponse().data.url);
	});

	it('should display error page when game data is incorrect', () => {
		setErrorResponse();
		let { getByTestId } = render(<Play />);

		expect(getByTestId('ErrorPage')).toBeInTheDocument();
	});
});
