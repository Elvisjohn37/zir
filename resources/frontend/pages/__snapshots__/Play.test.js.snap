// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Play /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="gamewindow"
    data-testid="LangDataProvider"
  >
    <div>
      gamewindow
    </div>
    <div
      class="header"
    >
      <div
        class="headerContent"
      >
        <div
          class="topContainer"
        >
          <div
            class="runningDateTime"
          >
            <div
              class="testRunningDateTime"
              data-testid="RunningDateTime"
            />
          </div>
        </div>
        <div
          class="bottomContainer"
        >
          <div
            class="announcementWrapper"
          >
            <div
              class="testAnnouncement"
              data-testid="Announcement"
              ismobile="false"
              viewtype="zy"
            />
          </div>
          <div
            class="balancesWrapper"
          >
            <div
              class="maxPayoutWrapper"
            >
              <div
                class="maxPayout"
              >
                <div
                  class="amountText"
                >
                  <span
                    class="amountFigure"
                  >
                    <div
                      data-testid="FormattedMessage"
                      id="maxPayoutPerBet"
                    >
                      <div>
                        maxPayoutPerBet
                      </div>
                    </div>
                    undefined 
                    <div
                      compactstart="100000000"
                      data-testid="MoneyFormatToggle"
                      value="700000000.000000"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="accountBalance"
              compactstart="100000000"
              data-testid="AccountBalance"
              format="{currencyCode} {balance}"
              initialbalanceinfo="{\\"playableBalance\\":\\"1457.800000\\",\\"currencyCode\\":\\"ZWD\\"}"
              isautoupdate="true"
              withrefreshbutton="true"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="content"
      >
        <div
          class="iframeSpinner"
          color="primary"
          data-testid="Spinner"
        >
          <div>
            Loading Spinner
          </div>
        </div>
        <iframe
          class="gameIframe"
          src="https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641"
        />
      </div>
    </div>
    <div
      data-testid="Dialog"
      maxwidth="xs"
    >
      <div
        data-testid="DialogContent"
      >
        <div
          classes="{\\"root\\":\\"zeroBalanceMessage\\"}"
          data-testid="DialogContentText"
        >
          <div
            data-testid="FormattedMessage"
            id="zeroBalance"
          >
            <div>
              zeroBalance
            </div>
          </div>
        </div>
        <div
          data-testid="DialogActions"
        >
          <button
            color="secondary"
            data-testid="Button"
            variant="contained"
          >
            <div
              data-testid="FormattedMessage"
              id="ok"
            >
              <div>
                ok
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Play /> should match snapshot 2`] = `
<DocumentFragment>
  <div
    category="gamewindow"
    data-testid="LangDataProvider"
  >
    <div>
      gamewindow
    </div>
    <div
      data-testid="Announcement"
      ismobile="true"
      viewtype="zy"
    />
    <div
      class="body mobileBody"
    >
      <div
        class="content"
      >
        <div
          class="iframeSpinner"
          color="primary"
          data-testid="Spinner"
        >
          <div>
            Loading Spinner
          </div>
        </div>
        <iframe
          class="gameIframe"
          src="https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641"
        />
      </div>
    </div>
    <div
      data-testid="Dialog"
      maxwidth="xs"
    >
      <div
        data-testid="DialogContent"
      >
        <div
          classes="{\\"root\\":\\"zeroBalanceMessage\\"}"
          data-testid="DialogContentText"
        >
          <div
            data-testid="FormattedMessage"
            id="zeroBalance"
          >
            <div>
              zeroBalance
            </div>
          </div>
        </div>
        <div
          data-testid="DialogActions"
        >
          <button
            color="secondary"
            data-testid="Button"
            variant="contained"
          >
            <div
              data-testid="FormattedMessage"
              id="ok"
            >
              <div>
                ok
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
